proto/viam/robot/v1/robot.proto

```diff
@@ -360,3 +367,33 @@ message RestartModuleResponse {}
 message ShutdownRequest {}
 
 message ShutdownResponse {}
+
+message GetMachineStatusRequest {}
+
+message GetMachineStatusResponse {
+  repeated ResourceStatus resources = 1;
+}
+
+message ResourceStatus {
+  enum State {
+    STATE_UNSPECIFIED = 0;
+    // a newly created resource.
+    STATE_UNCONFIGURED = 1;
+    // a resource that is being configured.
+    STATE_CONFIGURING = 2;
+    // a resource that has been successfully configured once and is not being
+    // re-configured or removed.
+    STATE_READY = 3;
+    // a resource that is being removed from the robot.
+    STATE_REMOVING = 4;
+  }
```

Comment:
I wonder if adding `STATE_UNHEALTHY` eventually will be a breaking change. Not the end of the world, but curious how enum modifications are handled (i.e. if the "indices" of the other variants don't change, maybe it's not breaking).

Reply:
This shouldn't be breaking based on my read of https://protobuf.dev/programming-guides/enum/#definitions. 

The key point is that `proto3` treats enums as "open", which means that values outside of the enum range are set directly on the field. So an old client can receive messages with this field containing values that are outside of the defined enum range.

Reply:
which is great, since we might add new states beyond `unhealthy` in the future

Reply:
Awesome, that's my reading, too.